system/lib/wasmfs/wasmfs.cpp

```diff
@@ -56,12 +56,17 @@ long __syscall_dup(long fd) {
 
   return fileTable.add(openFile.unlocked());
 }
-
-__wasi_errno_t __wasi_fd_write(__wasi_fd_t fd,
-                               const __wasi_ciovec_t* iovs,
-                               size_t iovs_len,
-                               __wasi_size_t* nwritten) {
-  if (iovs_len < 0) {
+enum class OffsetHandling { overwrite, preserve };
```

Comment:
How about `OpenFileState` and `Argument`? These control not just whether the offset in the open file state is overwritten, but also which offset is used in the first place.